activerecord/lib/active_record/relation/query_methods.rb

```diff
@@ -634,6 +628,13 @@ def where!(opts, *rest) # :nodoc:
       self
     end
 
+    def where_not!(opts, *rest) # :nodoc:
+      opts = sanitize_forbidden_attributes(opts)
+      references!(PredicateBuilder.references(opts)) if Hash === opts
+      self.where_clause += where_clause_factory.inverted.build(opts, rest)
+      self
+    end
+
```

Comment:
Since `where_not` is used only in one place, I'm not sure extracting it into it's own method adds anything (i.e easier to read, understand, or reuse)